source/common/access_log/access_log_manager_impl.cc

```diff
@@ -11,6 +11,14 @@
 namespace Envoy {
 namespace AccessLog {
 
+AccessLogManagerImpl::~AccessLogManagerImpl() {
+  for (auto& access_log : access_logs_) {
+    ENVOY_LOG(debug, "destroying access logger {}", access_log.first);
+    access_log.second = nullptr;
```

Comment:
nit: I think we prefer `access_log.second.reset()`

Also, since this is a shared ptr, I expect your shutdown delay would be elsewhere. I guess you'd see it in the listener manager logging you've added, but it seems like it would have to be access logger flush threads blocked writing, which (for posix) is a call to write (2).

Reply:
I think the [access_log_manager](https://github.com/envoyproxy/envoy/blob/229faebfca816ae34754146dd5a8d564e8944092/source/server/server.h#L332) will be deleted after the listener manager, so by the time the access log manager is being destroyed it should have the last reference to the access logs so this will involve actual object destruction.